### PR TITLE
Fix image vulnerabilities

### LINKED ISSUE #75: Vulnerabilities found by Aquasec
Aquasec scan found the following high or critical severity vulnerabilities in the image:
- GHSA-9763-4f94-gfch
- CVE-2024-41110
- GHSA-7jwh-3vrq-q3m8
- CVE-2024-27304
- CVE-2024-27289
- CVE-2024-27304
- CVE-2024-27289
- CVE-2023-45288
- CVE-2023-44487
- CVE-2023-39325
- CVE-2023-45288
- CVE-2024-24786
- CVE-2024-3596

### PR DESCRIPTION
**Pull request by K-Pomian**
Fix #75

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -14,7 +14,7 @@ require (
 	github.com/cenkalti/backoff/v3 v3.2.2 // indirect
 	github.com/davecgh/go-spew v1.1.2-0.20180830191138-d8f796af33cc // indirect
 	github.com/distribution/reference v0.6.0 // indirect
-	github.com/docker/docker v25.0.5+incompatible // indirect
+	github.com/docker/docker v25.0.6+incompatible // indirect
```
Comment (omordyk) [inline go.mod:17]: @K-Pomian pls rebase on my PR: https://github.com/open-horizon/vault-exchange-auth/pull/78
We need version of **github.com/docker/docker** no lower than v26.1.5.

**Review by omordyk — CHANGES_REQUESTED** [review]

**Comment by omordyk** [comment]
Should be merged after https://github.com/open-horizon/vault-exchange-auth/pull/78

File: Makefile [hunk 2]
```diff
@@ -4,7 +4,7 @@ SHELL := /bin/bash
 arch_tag ?= $(shell ./tools/arch-tag)
 arch ?= $(arch_tag)
 
-VAULT_VERSION ?= 1.14.8
+VAULT_VERSION ?= 1.14.13+ent
```
Comment (bencourliss) [inline Makefile:7]: I think we need to be at 1.14.10 non-enterprise version.

**Review by bencourliss — CHANGES_REQUESTED** [review]

**Comment by bencourliss** [comment]
> @bencourliss I recall that vault version we are locked for is v1.14.9?

We're locked to the minor version. So any patch of 1.14 is fine.

**Review by bencourliss — APPROVED** [review]
LGTM

**Comment by naphelps** [comment]
@K-Pomian This PR needs to be backed out of the history. Vault needs to be downgraded to and locked to 1.14.8.

**Comment by naphelps** [comment]
I fixed the source and history with a forced rebase. 8b702434a7b5478636b6ed613760a72d79c50879

The fixed container runs in a local all-in-one. The dockerhub testing tag has been updated. The version tag still needs to be rebuilt.

**Pull request merged**